src/components/state.tsx

```diff
@@ -0,0 +1,31 @@
+import { createContext, ReactNode, useState } from 'react';
+import { Person } from '../types';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const MyContext = createContext<any>(null);
+
+interface MyProvideProps {
+  children: ReactNode;
+}
+
+export const MyProvider = ({ children }: MyProvideProps) => {
+  const [person, setPerson] = useState<Person[]>([]);
+  const [error, setError] = useState('');
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [selectedPerson, setSelectedPerson] = useState<string | null>(null);
+
+  const contextValue = {
```

Comment:
The naming for the state `person` is not clear enough. It seems like it's an array of persons, so it would be better to name it `persons`.